### PR TITLE
feat(Unification): add internal log group api and consume in .within()

### LINKED ISSUE #20433: Add internal log group API
### What would you like?

To easily enabled log groups within commands.

### Why is this needed?

_No response_

### Other

_No response_

### PR DESCRIPTION
**Pull request by emilyrohrbough**
- Enhances # 20433 but a follow up ticket will be logged to determine/capture remaining work. This PR may be closed and merged with #20465 

### Summary
- Added log group api to be consumed internally by Cypress commands.
- Updated `.within()` to consume log group api
- Update Reporter's Command Group to be open by default for parent or child commands
- Fixed session's 'warn' state to be 'warning' to align with other 'warning' states.

### User facing changelog
Enhanced `.within()` to provide visual indication of nested commands and logs.

### How has the user experience changed?

Before | After
-- | --
Screen Shot 2022-03-04 at 4 36 44 PM| Screen Shot 2022-03-04 at 4 33 17 PM

### PR Tasks

- [x] Have tests been added/updated?
- [x] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [n/a] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [n/a] 	Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [n/a] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/driver/src/cy/logGroup.ts
Comment (emilyrohrbough) [inline packages/driver/src/cy/logGroup.ts:34]: When `Cypress.log` has [`emitOnly: true`](https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cypress/log.ts#L601) it will not send emit the log to the runner/reporter and will not return a log instance. It will however, create the log attributes and store it in the LogManager and add the logs on the Command. This seems to work just fine, however,

- Is there risk of data leaking when this test/command is sent to the dashbaord? I didn't find this attribute in the [schema](https://github.com/cypress-io/json-schemas/blob/master/src/schemas/test-runner-test.ts), but wanted to double check.
- Is there risk of this behaving unexpectedly if the user were to pass `{ log: false }` to the command consuming this?

For example:
```
cy.get('#form').within({ log: false}, () => {})
```
This example is likely less applicable to `.within()` and more applicable to up-coming consumption in `.withToDomain()`.

**Comment by brian-mann** [comment]
I think it visually looks super odd for the darkest background color not to extend the full width of the row. It's currently "indenting" with the caret. Is this intentional?

**Comment by emilyrohrbough** [comment]
@brian-mann That is the current designs of 10.0. The follow up PR to update the styles will remove this: #20465

**Comment by brian-mann** [comment]
I see, and the styles are expected to merge first right?

**Comment by emilyrohrbough** [comment]
It wouldn't really matter which order these went in. I figured merging this first might be better so it can unblock multi-domain. 🤷🏻‍♀️ Would you prefer the other way around?

File: packages/reporter/cypress/e2e/commands.cy.ts
Comment (flotwig) [inline packages/reporter/cypress/e2e/commands.cy.ts:360]: I love this change. No more command log peekaboo

File: packages/driver/src/cypress/log.ts
Comment (flotwig) [inline packages/driver/src/cypress/log.ts:40]: I think this should be combined somehow with the `LogConfig` defined in `internal-types.d.ts`:
https://github.com/cypress-io/cypress/blob/938f72ac3b2b03eb8112a52e28b32596d35d7ceb/packages/driver/types/internal-types.d.ts#L51-L70

Comment (flotwig) [inline packages/driver/src/cypress/log.ts:40]: I think this should be combined somehow with the `LogConfig` defined in `internal-types.d.ts`:
https://github.com/cypress-io/cypress/blob/938f72ac3b2b03eb8112a52e28b32596d35d7ceb/packages/driver/types/internal-types.d.ts#L51-L70

File: packages/driver/src/cy/logGroup.ts
Comment (flotwig) [inline packages/driver/src/cy/logGroup.ts:5]: doesn't this extend `LogConfig` since we pass it to `Cypress.log`?

Comment (flotwig) [inline packages/driver/src/cy/logGroup.ts:5]: doesn't this extend `LogConfig`?

**Review by flotwig — CHANGES_REQUESTED** [review]

**Review by tbiethman — DISMISSED** [review]
The API seems to work great; I wrote a few of my own tests featuring .within and they look how I'd expect at this stage.

**Comment by emilyrohrbough** [comment]
Moving to draft. This will go in after the command styles are merged.

File: packages/driver/src/cypress/log.ts
Comment (emilyrohrbough) [reply packages/driver/src/cypress/log.ts:40]: @flotwig I purposely limited these options to keep this API clean and minimal. 

I looked at adding them to interna-types.d.ts, but I'd prefer to keep the types with the code it's related to so it's easier to maintain and is face-up on the expected API. This seems to be the pattern within the 10.0-release branch so this isn't diverging from other files/packages.

File: packages/driver/src/cy/logGroup.ts
Comment (emilyrohrbough) [reply packages/driver/src/cy/logGroup.ts:5]: @flotwig I purposely limited these options to keep this API clean and minimal. As additional use-cases are determined, we can expand this API.

**Comment by emilyrohrbough** [comment]
Closing this PR as this logic is going into develop first in #20857. I'll open a new PR for consuming in the within command once we've landed on the new styles #20465 .

**Pull request closed without merging**
